src/main/java/net/rubygrapefruit/platform/file/FileWatchEvent.java

```diff
@@ -4,35 +4,21 @@
 
 public interface FileWatchEvent {
 
-    /**
-     * The type of the change. When {@link Type#FAILURE}, {@link #getFailure()} returns
-     * the failure, and {@link #getPath()} returns {@code null}.
-     * Otherwise {@link #getFailure()} returns {@code null} and {@link #getPath()}
-     * returns either the path of the change, or {@code null} if a path is not known.
-     */
-    Type getType();
+    void handleEvent(Handler handler);
```

Comment:
We could use the usual `accept(Visitor)` names. I am not sure `handle` is better. I would drop the `Event` from `handleEvent` for sure.

Reply:
The reason I prefer longer, less generic names is finding usages of `accept()` in a codebase the size of Gradle take long and turn up a number of false positives.